Installation redmine plugin fails

SCM-Manager 2.2.0 on Debian Buster (10 stable)

Installed readme plugin: success
Installed redmine plugin: not so much. See log

```
2020-07-03 18:04:08.273 [main] [          ] INFO  sonia.scm.lifecycle.BootstrapContextListener - start scm-manager initialization
2020-07-03 18:04:08.794 [main] [          ] INFO  sonia.scm.plugin.PluginProcessor - collect plugins
2020-07-03 18:04:09.004 [main] [          ] INFO  sonia.scm.plugin.PluginProcessor - install plugin tree:
+- scm-readme-plugin
+- scm-legacy-plugin
+- scm-svn-plugin
+- scm-hg-plugin
+- scm-redmine-plugin
+- scm-git-plugin
+- scm-issuetracker-plugin
   +- scm-redmine-plugin

2020-07-03 18:04:09.942 [main] [          ] INFO  sonia.scm.update.repository.XmlRepositoryV1UpdateStep - no v1 repositories database file found
2020-07-03 18:04:10.270 [main] [          ] INFO  sonia.scm.update.UpdateEngine - running update step for type sonia.scm.redmine.config.global.xml and version 2.0.0 (class sonia.scm.redmine.update.RedmineV2GlobalConfigMigrationUpdateStep)
2020-07-03 18:04:10.317 [main] [          ] INFO  sonia.scm.update.UpdateEngine - running update step for type sonia.scm.redmine.config.repository.xml and version 2.0.0 (class sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep)
2020-07-03 18:04:10.340 [main] [          ] ERROR sonia.scm.migration.UpdateException - could not execute update for type sonia.scm.redmine.config.repository.xml to version 2.0.0 in class sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep
java.lang.NullPointerException: null
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at java.base/java.util.stream.FindOps$FindSink$OfRef.get(FindOps.java:194)
	at java.base/java.util.stream.FindOps$FindSink$OfRef.get(FindOps.java:191)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at sonia.scm.update.V1Properties.getOptional(V1Properties.java:61)
	at sonia.scm.update.V1Properties.getBoolean(V1Properties.java:65)
	at sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep.buildConfig(RedmineV2ConfigMigrationUpdateStep.java:69)
	at sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep.lambda$doUpdate$0(RedmineV2ConfigMigrationUpdateStep.java:61)
	at sonia.scm.update.MapBasedPropertyReaderInstance.call(MapBasedPropertyReaderInstance.java:58)
	at sonia.scm.update.MapBasedPropertyReaderInstance.lambda$forEachEntry$0(MapBasedPropertyReaderInstance.java:44)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet.lambda$entryConsumer$0(Collections.java:1576)
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1746)
	at java.base/java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet$UnmodifiableEntrySetSpliterator.forEachRemaining(Collections.java:1601)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at sonia.scm.update.MapBasedPropertyReaderInstance.forEachEntry(MapBasedPropertyReaderInstance.java:44)
	at sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep.doUpdate(RedmineV2ConfigMigrationUpdateStep.java:61)
	at sonia.scm.update.UpdateEngine.execute(UpdateEngine.java:87)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at sonia.scm.update.UpdateEngine.update(UpdateEngine.java:77)
	at sonia.scm.lifecycle.BootstrapContextListener.processUpdates(BootstrapContextListener.java:175)
	at sonia.scm.lifecycle.BootstrapContextListener.startEitherMigrationOrApplication(BootstrapContextListener.java:133)
	at sonia.scm.lifecycle.BootstrapContextListener.createMigrationOrNormalModuleProvider(BootstrapContextListener.java:124)
	at sonia.scm.lifecycle.BootstrapContextListener.createStageTwoInjector(BootstrapContextListener.java:117)
	at sonia.scm.lifecycle.BootstrapContextListener.getInjector(BootstrapContextListener.java:93)
	at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:45)
	at sonia.scm.lifecycle.BootstrapContextListener.contextInitialized(BootstrapContextListener.java:75)
	at sonia.scm.lifecycle.BootstrapContextFilter.initializeContext(BootstrapContextFilter.java:74)
	at sonia.scm.lifecycle.BootstrapContextFilter.init(BootstrapContextFilter.java:65)
	at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:134)
	at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:749)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:773)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:360)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1445)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1409)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:844)
	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:275)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.server.Server.start(Server.java:408)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.server.Server.doStart(Server.java:372)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at sonia.scm.server.ScmServer.init(ScmServer.java:130)
	at sonia.scm.server.ScmServer.run(ScmServer.java:91)
	at sonia.scm.server.ScmServerDaemon.main(ScmServerDaemon.java:53)
2020-07-03 18:04:10.341 [main] [          ] ERROR sonia.scm.lifecycle.BootstrapContextListener - failed to create stage two injector
sonia.scm.migration.UpdateException: could not execute update for type sonia.scm.redmine.config.repository.xml to version 2.0.0 in class sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep
	at sonia.scm.update.UpdateEngine.execute(UpdateEngine.java:90)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at sonia.scm.update.UpdateEngine.update(UpdateEngine.java:77)
	at sonia.scm.lifecycle.BootstrapContextListener.processUpdates(BootstrapContextListener.java:175)
	at sonia.scm.lifecycle.BootstrapContextListener.startEitherMigrationOrApplication(BootstrapContextListener.java:133)
	at sonia.scm.lifecycle.BootstrapContextListener.createMigrationOrNormalModuleProvider(BootstrapContextListener.java:124)
	at sonia.scm.lifecycle.BootstrapContextListener.createStageTwoInjector(BootstrapContextListener.java:117)
	at sonia.scm.lifecycle.BootstrapContextListener.getInjector(BootstrapContextListener.java:93)
	at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:45)
	at sonia.scm.lifecycle.BootstrapContextListener.contextInitialized(BootstrapContextListener.java:75)
	at sonia.scm.lifecycle.BootstrapContextFilter.initializeContext(BootstrapContextFilter.java:74)
	at sonia.scm.lifecycle.BootstrapContextFilter.init(BootstrapContextFilter.java:65)
	at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:134)
	at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:749)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:773)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:360)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1445)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1409)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:844)
	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:275)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.server.Server.start(Server.java:408)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.server.Server.doStart(Server.java:372)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at sonia.scm.server.ScmServer.init(ScmServer.java:130)
	at sonia.scm.server.ScmServer.run(ScmServer.java:91)
	at sonia.scm.server.ScmServerDaemon.main(ScmServerDaemon.java:53)
Caused by: java.lang.NullPointerException: null
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at java.base/java.util.stream.FindOps$FindSink$OfRef.get(FindOps.java:194)
	at java.base/java.util.stream.FindOps$FindSink$OfRef.get(FindOps.java:191)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at sonia.scm.update.V1Properties.getOptional(V1Properties.java:61)
	at sonia.scm.update.V1Properties.getBoolean(V1Properties.java:65)
	at sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep.buildConfig(RedmineV2ConfigMigrationUpdateStep.java:69)
	at sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep.lambda$doUpdate$0(RedmineV2ConfigMigrationUpdateStep.java:61)
	at sonia.scm.update.MapBasedPropertyReaderInstance.call(MapBasedPropertyReaderInstance.java:58)
	at sonia.scm.update.MapBasedPropertyReaderInstance.lambda$forEachEntry$0(MapBasedPropertyReaderInstance.java:44)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet.lambda$entryConsumer$0(Collections.java:1576)
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1746)
	at java.base/java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet$UnmodifiableEntrySetSpliterator.forEachRemaining(Collections.java:1601)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at sonia.scm.update.MapBasedPropertyReaderInstance.forEachEntry(MapBasedPropertyReaderInstance.java:44)
	at sonia.scm.redmine.update.RedmineV2ConfigMigrationUpdateStep.doUpdate(RedmineV2ConfigMigrationUpdateStep.java:61)
	at sonia.scm.update.UpdateEngine.execute(UpdateEngine.java:87)
	... 45 common frames omitted

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - Fixed file extension detection with new spotter version
 - Fixed wrong cache directory location ([#1236](https://github.com/scm-manager/scm-manager/issues/1236) and [#1242](https://github.com/scm-manager/scm-manager/issues/1242))
+- Fixed error in update step ([#1237](https://github.com/scm-manager/scm-manager/issues/1237) and [#1244](https://github.com/scm-manager/scm-manager/issues/1244))
 
 ## [2.2.0] - 2020-07-03
 ### Added

--- a/scm-core/src/main/java/sonia/scm/update/V1Properties.java
+++ b/scm-core/src/main/java/sonia/scm/update/V1Properties.java
@@ -21,7 +21,7 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-    
+
 package sonia.scm.update;
 
 import javax.xml.bind.annotation.XmlAccessType;
@@ -58,7 +58,11 @@ public class V1Properties {
   }
 
   public Optional<String> getOptional(String key) {
-    return streamProps().filter(p -> key.equals(p.getKey())).map(V1Property::getValue).findFirst();
+    return streamProps()
+      .filter(prop -> prop.getValue() != null)
+      .filter(p -> key.equals(p.getKey()))
+      .map(V1Property::getValue)
+      .findFirst();
   }
 
   public Optional<Boolean> getBoolean(String key) {

--- a/scm-dao-xml/src/test/java/sonia/scm/update/xml/XmlV1PropertyDAOTest.java
+++ b/scm-dao-xml/src/test/java/sonia/scm/update/xml/XmlV1PropertyDAOTest.java
@@ -1,0 +1,162 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020-present Cloudogu GmbH and Contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package sonia.scm.update.xml;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import sonia.scm.SCMContextProvider;
+import sonia.scm.Stage;
+import sonia.scm.security.KeyGenerator;
+import sonia.scm.store.JAXBConfigurationEntryStoreFactory;
+import sonia.scm.update.RepositoryV1PropertyReader;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+
+class XmlV1PropertyDAOTest {
+
+  public static final String PROPERTIES = "<?xml version=\"1.0\" ?>\n" +
+    "<configuration>\n" +
+    "  <entry>\n" +
+    "    <key>9ZQKlvI401</key>\n" +
+    "    <value>\n" +
+    "      <item></item>\n" +
+    "      <item>\n" +
+    "        <key>redmine.url</key>\n" +
+    "        <value>https://redmine.example.net/projects/r6</value>\n" +
+    "      </item>\n" +
+    "      <item>\n" +
+    "        <key>redmine.auto-close-username-transformer</key>\n" +
+    "        <value>{0}</value>\n" +
+    "      </item>\n" +
+    "      <item>\n" +
+    "        <key>redmine.auto-close</key>\n" +
+    "      </item>\n" +
+    "    </value>\n" +
+    "  </entry>\n" +
+    "  <entry>\n" +
+    "    <key>1wPsrHPM81</key>\n" +
+    "    <value>\n" +
+    "      <item>\n" +
+    "        <key>notify.contact.repository</key>\n" +
+    "        <value>true</value>\n" +
+    "      </item>\n" +
+    "      <item>\n" +
+    "        <key>redmine.auto-close-username-transformer</key>\n" +
+    "        <value>fixed, fix, closed, close, resolved, resolve</value>\n" +
+    "      </item>\n" +
+    "      <item>\n" +
+    "        <key>redmine.url</key>\n" +
+    "        <value>https://redmine.example.net/projects/a2</value>\n" +
+    "      </item>\n" +
+    "      <item>\n" +
+    "        <key>redmine.update-issues</key>\n" +
+    "        <value>true</value>\n" +
+    "      </item>\n" +
+    "      <item>\n" +
+    "        <key>notify.email.per.push</key>\n" +
+    "        <value>true</value>\n" +
+    "      </item>\n" +
+    "      <item>\n" +
+    "        <key>notify.use.author.as.from.address</key>\n" +
+    "        <value>true</value>\n" +
+    "      </item>\n" +
+    "      <item>\n" +
+    "        <key>redmine.auto-close</key>\n" +
+    "        <value>true</value>\n" +
+    "      </item>\n" +
+    "    </value>\n" +
+    "  </entry>\n" +
+    "  <entry>\n" +
+    "    <key>WlDszQtZj4</key>\n" +
+    "    <value></value>\n" +
+    "  </entry>\n" +
+    "</configuration>";
+
+  /**
+   * This is a test for https://github.com/scm-manager/scm-manager/issues/1237
+   */
+  @Test
+  void shouldReadItemsWithEmptyValues(@TempDir Path temp) throws IOException {
+    Path configPath = temp.resolve("config");
+    Files.createDirectories(configPath);
+    Path propFile = configPath.resolve("repository-properties-v1.xml");
+    Files.write(propFile, PROPERTIES.getBytes());
+    XmlV1PropertyDAO dao = new XmlV1PropertyDAO(new JAXBConfigurationEntryStoreFactory(new SimpleContextProvider(temp), null, new SimpleKeyGenerator()));
+
+    dao.getProperties(new RepositoryV1PropertyReader())
+      .forEachEntry((key, prop) -> {
+        if (key.equals("9ZQKlvI401")) {
+          Assertions.assertThat(prop.getBoolean("redmine.auto-close")).isNotPresent();
+        } else if (key.equals("1wPsrHPM81")) {
+          Assertions.assertThat(prop.getBoolean("redmine.auto-close")).isPresent().get().isEqualTo(true);
+        }
+      });
+  }
+
+  private static class SimpleContextProvider implements SCMContextProvider {
+    private final Path temp;
+
+    public SimpleContextProvider(Path temp) {
+      this.temp = temp;
+    }
+
+    @Override
+    public File getBaseDirectory() {
+      return temp.toFile();
+    }
+
+    @Override
+    public Path resolve(Path path) {
+      return null;
+    }
+
+    @Override
+    public Stage getStage() {
+      return null;
+    }
+
+    @Override
+    public Throwable getStartupError() {
+      return null;
+    }
+
+    @Override
+    public String getVersion() {
+      return null;
+    }
+  }
+
+  private static class SimpleKeyGenerator implements KeyGenerator {
+    @Override
+    public String createKey() {
+      return "" + System.nanoTime();
+    }
+  }
+}

